docs/cookbook.rst

```diff
@@ -0,0 +1,70 @@
+Heksher-py Cookbook
+---------------------------------
+
+.. _renaming a setting:
+
+Renaming a setting
+====================
+
+Say you have a setting called ``foo``
+
+.. code-block:: python
+
+    foo = Setting('foo', int, ...)
+
+And you want to change the name of the setting to ``bar``. If you were to simply change the name of the setting, you
+will run into the problem of essentially having declared a new setting. This means that all the rules defined in heksher
+for ``foo`` will not transfer over to ``bar``.
+
+To fix this, we'll use the "alias" parameter when declaring the setting.
+
+.. code-block:: python
+
+    bar = Setting('bar', int, ..., alias='foo')
+
+This will inform the heksher service that the setting known as ``foo`` should be renamed to bar, along with all the
+rules associated with it. As an added bonus, any app that uses the older name will not break, as they will be able to
+access ``bar`` via its alias (and old name) ``foo``.
```

Comment:
what about multiple aliases?

Reply:
not supported

Reply:
I think you should write it, as well as if you can change a third time (by "aliasing" the new name)?